catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchSource.java

```diff
@@ -224,8 +235,7 @@ public void init() {
     configureXmlInputFactory();
   }
 
-  // lazy init
-  private SecureCxfClientFactory<OpenSearch> getClientFactory() {
+  private synchronized SecureCxfClientFactory<OpenSearch> getClientFactory() {
```

Comment:
❗️ Because you are synchronizing here, you need to make sure that later on when `factory` is accessed directly or updated as it is in a lot of setters, that those are synchronized as well.

Reply:
can I just remove the synchronized modifier?

Reply:
You still have to account for concurrency since you are doing lazy initialization, clearing, and accessing from different threads. 

It could potentially be replaced with `volatile` in this case.